File: generators/yetiNames.py
__all__ = ['yetiNames']

# Don't look below, you will not understand this Python code :) I don't.

from js2py.pyjs import *
# setting scope
var = Scope( JS_BUILTINS )
set_global_object(var)

# Code follows:
var.registers(['nm1', 'nm11', 'nm12', 'nm4', 'nameGen', 'nm5', 'nm8', 'nm14', 'nm3', 'nm4b', 'nm7', 'nm9', 'nm10', 'nm13', 'nm2', 'nm6'])
@Js
def PyJsHoisted_nameGen_(type, this, arguments, var=var):
    var = Scope({'arguments':arguments, 'this':this, 'type':type}, var)
    var.registers(['tp', 'result', 'type'])
    var.put('tp', var.get('type'))
    var.put('result', Js([]))
    #for JS loop
    var.put('i', Js(0.0))
    while (var.get('i')<Js(10.0)):
        try:
            if PyJsStrictEq(var.get('tp'),Js(1.0)):
                var.put('rnd', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm5').get('length'))))
                var.put('rnd2', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm6').get('length'))))
                var.put('rnd3', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm7').get('length'))))
                var.put('rnd4', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm6').get('length'))))
                var.put('rnd5', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm8').get('length'))))
                if (var.get('i')<Js(6.0)):
                    while PyJsStrictEq(var.get('nm5').get(var.get('rnd')),var.get('nm7').get(var.get('rnd3'))):
                        var.put('rnd3', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm7').get('length'))))
                    if (var.get('rnd')<Js(5.0)):
                        while ((var.get('rnd5')<Js(15.0)) or PyJsStrictEq(var.get('nm7').get(var.get('rnd3')),var.get('nm8').get(var.get('rnd5')))):
                            var.put('rnd5', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm8').get('length'))))
                    var.put('names', ((((var.get('nm5').get(var.get('rnd'))+var.get('nm6').get(var.get('rnd2')))+var.get('nm7').get(var.get('rnd3')))+var.get('nm6').get(var.get('rnd4')))+var.get('nm8').get(var.get('rnd5'))))
                else:
                    if (var.get('i')<Js(8.0)):
                        var.put('rnd6', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm9').get('length'))))
                        var.put('rnd7', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm6').get('length'))))
                        while PyJsStrictEq(var.get('nm5').get(var.get('rnd')),var.get('nm7').get(var.get('rnd3'))):
                            var.put('rnd3', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm7').get('length'))))
                        while PyJsStrictEq(var.get('nm9').get(var.get('rnd6')),var.get('nm8').get(var.get('rnd5'))):
                            var.put('rnd5', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm8').get('length'))))
                        while PyJsStrictEq(var.get('nm9').get(var.get('rnd6')),var.get('nm7').get(var.get('rnd3'))):
                            var.put('rnd6', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm9').get('length'))))
                        var.put('names', ((((((var.get('nm5').get(var.get('rnd'))+var.get('nm6').get(var.get('rnd2')))+var.get('nm7').get(var.get('rnd3')))+var.get('nm6').get(var.get('rnd4')))+var.get('nm9').get(var.get('rnd6')))+var.get('nm6').get(var.get('rnd7')))+var.get('nm8').get(var.get('rnd5'))))
                    else:
                        var.put('rnd6', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm9').get('length'))))
                        var.put('rnd7', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm6').get('length'))))
                        while PyJsStrictEq(var.get('nm5').get(var.get('rnd')),var.get('nm7').get(var.get('rnd3'))):
                            var.put('rnd3', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm7').get('length'))))
                        while PyJsStrictEq(var.get('nm7').get(var.get('rnd3')),var.get('nm8').get(var.get('rnd5'))):
                            var.put('rnd5', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm8').get('length'))))
                        while PyJsStrictEq(var.get('nm9').get(var.get('rnd6')),var.get('nm7').get(var.get('rnd3'))):
                            var.put('rnd6', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm9').get('length'))))
                        var.put('names', ((((((var.get('nm5').get(var.get('rnd'))+var.get('nm6').get(var.get('rnd2')))+var.get('nm9').get(var.get('rnd6')))+var.get('nm6').get(var.get('rnd7')))+var.get('nm7').get(var.get('rnd3')))+var.get('nm6').get(var.get('rnd4')))+var.get('nm8').get(var.get('rnd5'))))
            else:
                if PyJsStrictEq(var.get('tp'),Js(2.0)):
                    var.put('rnd', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm10').get('length'))))
                    var.put('rnd2', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm11').get('length'))))
                    var.put('rnd3', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm12').get('length'))))
                    var.put('rnd4', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm11').get('length'))))
                    var.put('rnd5', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm13').get('length'))))
                    if (var.get('i')<Js(6.0)):
                        while PyJsStrictEq(var.get('nm10').get(var.get('rnd')),var.get('nm12').get(var.get('rnd3'))):
                            var.put('rnd3', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm12').get('length'))))
                        if (var.get('rnd')<Js(5.0)):
                            while ((var.get('rnd5')<Js(15.0)) or PyJsStrictEq(var.get('nm12').get(var.get('rnd3')),var.get('nm13').get(var.get('rnd5')))):
                                var.put('rnd5', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm13').get('length'))))
                        var.put('names', ((((var.get('nm10').get(var.get('rnd'))+var.get('nm11').get(var.get('rnd2')))+var.get('nm12').get(var.get('rnd3')))+var.get('nm11').get(var.get('rnd4')))+var.get('nm13').get(var.get('rnd5'))))
                    else:
                        if (var.get('i')<Js(8.0)):
                            var.put('rnd6', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm14').get('length'))))
                            var.put('rnd7', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm11').get('length'))))
                            while PyJsStrictEq(var.get('nm10').get(var.get('rnd')),var.get('nm12').get(var.get('rnd3'))):
                                var.put('rnd3', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm12').get('length'))))
                            while PyJsStrictEq(var.get('nm14').get(var.get('rnd6')),var.get('nm13').get(var.get('rnd5'))):
                                var.put('rnd5', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm13').get('length'))))
                            while PyJsStrictEq(var.get('nm14').get(var.get('rnd6')),var.get('nm12').get(var.get('rnd3'))):
                                var.put('rnd6', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm14').get('length'))))
                            var.put('names', ((((((var.get('nm10').get(var.get('rnd'))+var.get('nm11').get(var.get('rnd2')))+var.get('nm12').get(var.get('rnd3')))+var.get('nm11').get(var.get('rnd4')))+var.get('nm14').get(var.get('rnd6')))+var.get('nm11').get(var.get('rnd7')))+var.get('nm13').get(var.get('rnd5'))))
                        else:
                            var.put('rnd6', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm14').get('length'))))
                            var.put('rnd7', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm11').get('length'))))
                            while PyJsStrictEq(var.get('nm10').get(var.get('rnd')),var.get('nm12').get(var.get('rnd3'))):
                                var.put('rnd3', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm12').get('length'))))
                            while PyJsStrictEq(var.get('nm14').get(var.get('rnd6')),var.get('nm13').get(var.get('rnd5'))):
                                var.put('rnd5', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm13').get('length'))))
                            while PyJsStrictEq(var.get('nm14').get(var.get('rnd6')),var.get('nm12').get(var.get('rnd3'))):
                                var.put('rnd6', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm14').get('length'))))
                            var.put('names', ((((((var.get('nm10').get(var.get('rnd'))+var.get('nm11').get(var.get('rnd2')))+var.get('nm14').get(var.get('rnd6')))+var.get('nm11').get(var.get('rnd7')))+var.get('nm12').get(var.get('rnd3')))+var.get('nm11').get(var.get('rnd4')))+var.get('nm13').get(var.get('rnd5'))))
                else:
                    var.put('rnd', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm1').get('length'))))
                    var.put('rnd2', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm2').get('length'))))
                    var.put('rnd3', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm3').get('length'))))
                    var.put('rnd4', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm2').get('length'))))
                    var.put('rnd5', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm4').get('length'))))
                    if (var.get('i')<Js(5.0)):
                        while PyJsStrictEq(var.get('nm1').get(var.get('rnd')),var.get('nm3').get(var.get('rnd3'))):
                            var.put('rnd3', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm3').get('length'))))
                        if (var.get('rnd')<Js(5.0)):
                            while ((var.get('rnd5')<Js(2.0)) or PyJsStrictEq(var.get('nm3').get(var.get('rnd3')),var.get('nm4').get(var.get('rnd5')))):
                                var.put('rnd5', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm4').get('length'))))
                        var.put('names', ((((var.get('nm1').get(var.get('rnd'))+var.get('nm2').get(var.get('rnd2')))+var.get('nm3').get(var.get('rnd3')))+var.get('nm2').get(var.get('rnd4')))+var.get('nm4').get(var.get('rnd5'))))
                    else:
                        if (var.get('i')<Js(8.0)):
                            var.put('rnd6', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm4b').get('length'))))
                            var.put('rnd7', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm2').get('length'))))
                            while PyJsStrictEq(var.get('nm1').get(var.get('rnd')),var.get('nm3').get(var.get('rnd3'))):
                                var.put('rnd3', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm3').get('length'))))
                            while PyJsStrictEq(var.get('nm4b').get(var.get('rnd6')),var.get('nm4').get(var.get('rnd5'))):
                                var.put('rnd5', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm4').get('length'))))
                            while PyJsStrictEq(var.get('nm4b').get(var.get('rnd6')),var.get('nm3').get(var.get('rnd3'))):
                                var.put('rnd6', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm4b').get('length'))))
                            var.put('names', ((((((var.get('nm1').get(var.get('rnd'))+var.get('nm2').get(var.get('rnd2')))+var.get('nm3').get(var.get('rnd3')))+var.get('nm2').get(var.get('rnd4')))+var.get('nm4b').get(var.get('rnd6')))+var.get('nm2').get(var.get('rnd7')))+var.get('nm4').get(var.get('rnd5'))))
                        else:
                            var.put('rnd6', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm4b').get('length'))))
                            var.put('rnd7', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm2').get('length'))))
                            while PyJsStrictEq(var.get('nm1').get(var.get('rnd')),var.get('nm3').get(var.get('rnd3'))):
                                var.put('rnd3', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm3').get('length'))))
                            while PyJsStrictEq(var.get('nm3').get(var.get('rnd3')),var.get('nm4').get(var.get('rnd5'))):
                                var.put('rnd5', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm4').get('length'))))
                            while PyJsStrictEq(var.get('nm4b').get(var.get('rnd6')),var.get('nm3').get(var.get('rnd3'))):
                                var.put('rnd6', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm4b').get('length'))))
                            var.put('names', ((((((var.get('nm1').get(var.get('rnd'))+var.get('nm2').get(var.get('rnd2')))+var.get('nm4b').get(var.get('rnd6')))+var.get('nm2').get(var.get('rnd7')))+var.get('nm3').get(var.get('rnd3')))+var.get('nm2').get(var.get('rnd4')))+var.get('nm4').get(var.get('rnd5'))))
            var.get('result').callprop('push', var.get('names'))
        finally:
                (var.put('i',Js(var.get('i').to_number())+Js(1))-Js(1))
    pass
    return var.get('result')
PyJsHoisted_nameGen_.func_name = 'nameGen'
var.put('nameGen', PyJsHoisted_nameGen_)
var.put('nm1', Js([Js(''), Js(''), Js(''), Js(''), Js(''), Js('b'), Js('ch'), Js('dz'), Js('dj'), Js('g'), Js('gy'), Js('h'), Js('j'), Js('k'), Js('ky'), Js('l'), Js('m'), Js('n'), Js('r'), Js('s'), Js('sz'), Js('sh'), Js('t'), Js('th'), Js('v'), Js('z')]))
var.put('nm2', Js([Js('a'), Js('e'), Js('i'), Js('o'), Js('u'), Js('a'), Js('o'), Js('u')]))
var.put('nm3', Js([Js('dd'), Js('h'), Js('k'), Js('l'), Js('ll'), Js('m'), Js('n'), Js('r'), Js('s'), Js('t'), Js('z'), Js('dd'), Js('h'), Js('k'), Js('kk'), Js('l'), Js('ll'), Js('lr'), Js('m'), Js('n'), Js('ng'), Js('nr'), Js('nz'), Js('r'), Js('rr'), Js('s'), Js('sh'), Js('sr'), Js('t'), Js('ts'), Js('vr'), Js('y'), Js('z')]))
var.put('nm4', Js([Js(''), Js(''), Js('c'), Js('hn'), Js('hl'), Js('l'), Js('m'), Js('n'), Js('ng'), Js('r'), Js('rs'), Js('s'), Js('sh')]))
var.put('nm4b', Js([Js('h'), Js('k'), Js('l'), Js('ll'), Js('m'), Js('n'), Js('r'), Js('s'), Js('t'), Js('z')]))
var.put('nm5', Js([Js(''), Js(''), Js(''), Js(''), Js(''), Js('f'), Js('gy'), Js('h'), Js('k'), Js('ky'), Js('l'), Js('m'), Js('n'), Js('ph'), Js('r'), Js('s'), Js('sh'), Js('th'), Js('w'), Js('y'), Js('z')]))
var.put('nm6', Js([Js('a'), Js('e'), Js('i'), Js('o'), Js('u'), Js('a'), Js('e'), Js('i')]))
var.put('nm7', Js([Js('b'), Js('f'), Js('g'), Js('h'), Js('k'), Js('l'), Js('ll'), Js('m'), Js('n'), Js('r'), Js('rr'), Js('s'), Js('t'), Js('w'), Js('z'), Js('b'), Js('f'), Js('g'), Js('gy'), Js('h'), Js('hn'), Js('hl'), Js('k'), Js('l'), Js('ll'), Js('m'), Js('n'), Js('ng'), Js('nn'), Js('r'), Js('rr'), Js('rl'), Js('ry'), Js('rs'), Js('s'), Js('sh'), Js('t'), Js('ty'), Js('th'), Js('w'), Js('z')]))
var.put('nm8', Js([Js(''), Js(''), Js(''), Js(''), Js(''), Js(''), Js(''), Js(''), Js(''), Js(''), Js(''), Js(''), Js(''), Js(''), Js(''), Js('h'), Js('hn'), Js('l'), Js('m'), Js('n'), Js('ng'), Js('ph'), Js('s'), Js('sh'), Js('th'), Js('y')]))
var.put('nm9', Js([Js('b'), Js('f'), Js('g'), Js('h'), Js('k'), Js('l'), Js('ll'), Js('m'), Js('n'), Js('r'), Js('rr'), Js('s'), Js('t'), Js('w'), Js('z')]))
var.put('nm10', Js([Js(''), Js(''), Js(''), Js(''), Js(''), Js(''), Js('f'), Js('g'), Js('gy'), Js('h'), Js('k'), Js('ky'), Js('l'), Js('m'), Js('n'), Js('ph'), Js('r'), Js('s'), Js('sh'), Js('th'), Js('v'), Js('w'), Js('y'), Js('z')]))
var.put('nm11', Js([Js('a'), Js('e'), Js('i'), Js('o'), Js('u'), Js('a')]))
var.put('nm12', Js([Js('b'), Js('dd'), Js('g'), Js('h'), Js('k'), Js('kk'), Js('l'), Js('ll'), Js('m'), Js('n'), Js('nn'), Js('r'), Js('rr'), Js('s'), Js('t'), Js('w'), Js('y'), Js('z'), Js('b'), Js('dd'), Js('g'), Js('gy'), Js('h'), Js('hn'), Js('k'), Js('kk'), Js('l'), Js('ll'), Js('lr'), Js('m'), Js('n'), Js('ng'), Js('nn'), Js('nr'), Js('r'), Js('rr'), Js('ry'), Js('s'), Js('sh'), Js('sr'), Js('t'), Js('th'), Js('ts'), Js('vr'), Js('w'), Js('y'), Js('z')]))
var.put('nm13', Js([Js(''), Js(''), Js(''), Js(''), Js(''), Js(''), Js(''), Js(''), Js('h'), Js('hl'), Js('hn'), Js('l'), Js('m'), Js('n'), Js('ng'), Js('ph'), Js('r'), Js('rs'), Js('s'), Js('sh'), Js('th')]))
var.put('nm14', Js([Js('b'), Js('dd'), Js('g'), Js('h'), Js('k'), Js('kk'), Js('l'), Js('ll'), Js('m'), Js('n'), Js('nn'), Js('r'), Js('rr'), Js('s'), Js('t'), Js('w'), Js('y'), Js('z')]))
pass
pass


# Add lib to the module scope
yetiNames = var.to_python()

File: js/yetiNames.js
var nm1 = ["","","","","","b","ch","dz","dj","g","gy","h","j","k","ky","l","m","n","r","s","sz","sh","t","th","v","z"];
var nm2 = ["a","e","i","o","u","a","o","u"];
var nm3 = ["dd","h","k","l","ll","m","n","r","s","t","z","dd","h","k","kk","l","ll","lr","m","n","ng","nr","nz","r","rr","s","sh","sr","t","ts","vr","y","z"];
var nm4 = ["","","c","hn","hl","l","m","n","ng","r","rs","s","sh"];
var nm4b = ["h","k","l","ll","m","n","r","s","t","z"];

var nm5 = ["","","","","","f","gy","h","k","ky","l","m","n","ph","r","s","sh","th","w","y","z"];
var nm6 = ["a","e","i","o","u","a","e","i"];
var nm7 = ["b","f","g","h","k","l","ll","m","n","r","rr","s","t","w","z","b","f","g","gy","h","hn","hl","k","l","ll","m","n","ng","nn","r","rr","rl","ry","rs","s","sh","t","ty","th","w","z"];
var nm8 = ["","","","","","","","","","","","","","","","h","hn","l","m","n","ng","ph","s","sh","th","y"];
var nm9 = ["b","f","g","h","k","l","ll","m","n","r","rr","s","t","w","z"];

var nm10 = ["","","","","","","f","g","gy","h","k","ky","l","m","n","ph","r","s","sh","th","v","w","y","z"];
var nm11 = ["a","e","i","o","u","a"];
var nm12 = ["b","dd","g","h","k","kk","l","ll","m","n","nn","r","rr","s","t","w","y","z","b","dd","g","gy","h","hn","k","kk","l","ll","lr","m","n","ng","nn","nr","r","rr","ry","s","sh","sr","t","th","ts","vr","w","y","z"];
var nm13 = ["","","","","","","","","h","hl","hn","l","m","n","ng","ph","r","rs","s","sh","th"];
var nm14 = ["b","dd","g","h","k","kk","l","ll","m","n","nn","r","rr","s","t","w","y","z"];

var br = "";

function nameGen(type){
	$('#placeholder').css('textTransform', 'capitalize');
	var tp = type;
	var element = document.createElement("div");
	element.setAttribute("id", "result");
	
	for(i = 0; i < 10; i++){
		if(tp === 1){
			rnd = Math.floor(Math.random() * nm5.length);
			rnd2 = Math.floor(Math.random() * nm6.length);
			rnd3 = Math.floor(Math.random() * nm7.length);
			rnd4 = Math.floor(Math.random() * nm6.length);
			rnd5 = Math.floor(Math.random() * nm8.length);
			if(i < 6){
				while(nm5[rnd] === nm7[rnd3]){
					rnd3 = Math.floor(Math.random() * nm7.length);
				}
				if(rnd < 5){
					while(rnd5 < 15 || nm7[rnd3] === nm8[rnd5]){
						rnd5 = Math.floor(Math.random() * nm8.length);
					}
				}
				names = nm5[rnd] + nm6[rnd2] + nm7[rnd3] + nm6[rnd4] + nm8[rnd5];
			}else if(i < 8){
				rnd6 = Math.floor(Math.random() * nm9.length);
				rnd7 = Math.floor(Math.random() * nm6.length);
				while(nm5[rnd] === nm7[rnd3]){
					rnd3 = Math.floor(Math.random() * nm7.length);
				}
				while(nm9[rnd6] === nm8[rnd5]){
					rnd5 = Math.floor(Math.random() * nm8.length);
				}
				while(nm9[rnd6] === nm7[rnd3]){
					rnd6 = Math.floor(Math.random() * nm9.length);
				}
				names = nm5[rnd] + nm6[rnd2] + nm7[rnd3] + nm6[rnd4] + nm9[rnd6] + nm6[rnd7] + nm8[rnd5];
			}else{
				rnd6 = Math.floor(Math.random() * nm9.length);
				rnd7 = Math.floor(Math.random() * nm6.length);
				while(nm5[rnd] === nm7[rnd3]){
					rnd3 = Math.floor(Math.random() * nm7.length);
				}
				while(nm7[rnd3] === nm8[rnd5]){
					rnd5 = Math.floor(Math.random() * nm8.length);
				}
				while(nm9[rnd6] === nm7[rnd3]){
					rnd6 = Math.floor(Math.random() * nm9.length);
				}
				names = nm5[rnd] + nm6[rnd2] + nm9[rnd6] + nm6[rnd7] + nm7[rnd3] + nm6[rnd4] + nm8[rnd5];
			}
		}else if(tp === 2){
			rnd = Math.floor(Math.random() * nm10.length);
			rnd2 = Math.floor(Math.random() * nm11.length);
			rnd3 = Math.floor(Math.random() * nm12.length);
			rnd4 = Math.floor(Math.random() * nm11.length);
			rnd5 = Math.floor(Math.random() * nm13.length);
			if(i < 6){
				while(nm10[rnd] === nm12[rnd3]){
					rnd3 = Math.floor(Math.random() * nm12.length);
				}
				if(rnd < 5){
					while(rnd5 < 15 || nm12[rnd3] === nm13[rnd5]){
						rnd5 = Math.floor(Math.random() * nm13.length);
					}
				}
				names = nm10[rnd] + nm11[rnd2] + nm12[rnd3] + nm11[rnd4] + nm13[rnd5];
			}else if(i < 8){
				rnd6 = Math.floor(Math.random() * nm14.length);
				rnd7 = Math.floor(Math.random() * nm11.length);
				while(nm10[rnd] === nm12[rnd3]){
					rnd3 = Math.floor(Math.random() * nm12.length);
				}
				while(nm14[rnd6] === nm13[rnd5]){
					rnd5 = Math.floor(Math.random() * nm13.length);
				}
				while(nm14[rnd6] === nm12[rnd3]){
					rnd6 = Math.floor(Math.random() * nm14.length);
				}
				names = nm10[rnd] + nm11[rnd2] + nm12[rnd3] + nm11[rnd4] + nm14[rnd6] + nm11[rnd7] + nm13[rnd5];
			}else{
				rnd6 = Math.floor(Math.random() * nm14.length);
				rnd7 = Math.floor(Math.random() * nm11.length);
				while(nm10[rnd] === nm12[rnd3]){
					rnd3 = Math.floor(Math.random() * nm12.length);
				}
				while(nm14[rnd6] === nm13[rnd5]){
					rnd5 = Math.floor(Math.random() * nm13.length);
				}
				while(nm14[rnd6] === nm12[rnd3]){
					rnd6 = Math.floor(Math.random() * nm14.length);
				}
				names = nm10[rnd] + nm11[rnd2] + nm14[rnd6] + nm11[rnd7] + nm12[rnd3] + nm11[rnd4] + nm13[rnd5];
			}
		}else{
			rnd = Math.floor(Math.random() * nm1.length);
			rnd2 = Math.floor(Math.random() * nm2.length);
			rnd3 = Math.floor(Math.random() * nm3.length);
			rnd4 = Math.floor(Math.random() * nm2.length);
			rnd5 = Math.floor(Math.random() * nm4.length);
			if(i < 5){
				while(nm1[rnd] === nm3[rnd3]){
					rnd3 = Math.floor(Math.random() * nm3.length);
				}
				if(rnd < 5){
					while(rnd5 < 2 || nm3[rnd3] === nm4[rnd5]){
						rnd5 = Math.floor(Math.random() * nm4.length);
					}
				}
				names = nm1[rnd] + nm2[rnd2] + nm3[rnd3] + nm2[rnd4] + nm4[rnd5];
			}else if(i < 8){
				rnd6 = Math.floor(Math.random() * nm4b.length);
				rnd7 = Math.floor(Math.random() * nm2.length);
				while(nm1[rnd] === nm3[rnd3]){
					rnd3 = Math.floor(Math.random() * nm3.length);
				}
				while(nm4b[rnd6] === nm4[rnd5]){
					rnd5 = Math.floor(Math.random() * nm4.length);
				}
				while(nm4b[rnd6] === nm3[rnd3]){
					rnd6 = Math.floor(Math.random() * nm4b.length);
				}
				names = nm1[rnd] + nm2[rnd2] + nm3[rnd3] + nm2[rnd4] + nm4b[rnd6] + nm2[rnd7] + nm4[rnd5];
			}else{
				rnd6 = Math.floor(Math.random() * nm4b.length);
				rnd7 = Math.floor(Math.random() * nm2.length);
				while(nm1[rnd] === nm3[rnd3]){
					rnd3 = Math.floor(Math.random() * nm3.length);
				}
				while(nm3[rnd3] === nm4[rnd5]){
					rnd5 = Math.floor(Math.random() * nm4.length);
				}
				while(nm4b[rnd6] === nm3[rnd3]){
					rnd6 = Math.floor(Math.random() * nm4b.length);
				}
				names = nm1[rnd] + nm2[rnd2] + nm4b[rnd6] + nm2[rnd7] + nm3[rnd3] + nm2[rnd4] + nm4[rnd5];
			}
		}
		br = document.createElement('br');	
		element.appendChild(document.createTextNode(names));
		element.appendChild(br);
	}
	if(document.getElementById("result")){
		document.getElementById("placeholder").removeChild(document.getElementById("result"));
	}		
	document.getElementById("placeholder").appendChild(element);
}